cpp/include/ray/api/ray_remote.h

```diff
@@ -18,21 +18,59 @@
 
 namespace ray {
 
-namespace {
+namespace internal {
+
+inline static std::vector<absl::string_view> GetFunctionNames(absl::string_view str) {
+  std::vector<absl::string_view> output;
+  size_t first = 0;
+
+  while (first < str.size()) {
+    auto second = str.find_first_of(",", first);
+
+    if (first != second) {
+      auto sub_str = str.substr(first, second - first);
+      if (sub_str.find_first_of('(') != absl::string_view::npos) {
+        second = str.find_first_of(")", first) + 1;
+      }
+      if (str[first] == ' ') {
```

Comment:
macro will reduce all whitespaces to a single whitespace. newlines will become one line after macro expand.
if fill foo with other character will be treated as a concat string, only comma can be used to split arguments. 

It will be improved to handle the possible single whitespace in the tail.